### PR TITLE
Update

### PR DESCRIPTION
**Pull request by SuperDJ**
- Updated composer packages 
   - https://github.com/pusher/push-notifications-php/issues/37; 
   - https://github.com/pusher/push-notifications-php/pull/38. firebase/php-jwt v6 introduced breaking changes so just changing the version numbers isn't enough;
   - Updated guzzle to at least 7 as there security issues: https://packagist.org/packages/guzzlehttp/guzzle;
- Dropped support for old PHP versions https://www.php.net/supported-versions.php;
- Added files and .github actions from https://github.com/pusher/pusher-http-php (actions would still need some changes of someone with the know how);
- Dropped Travis CI (README needs to be updated);
- Added types where possible;

### CODE REVIEW

**Comment by benjamin-tang-pusher** [comment]
I tried to run your repo but when I try to trigger a message I get an error that the Guzzle client doesn't have a request method:
```
PHP Fatal error:  Uncaught Error: Call to a member function request() on null in /Users/benjamintang/Documents/development/php/pure/push-notifications-php-1/src/PushNotifications.php:65
Stack trace:
#0 /Users/benjamintang/Documents/development/php/pure/push-notifications-php-1/src/PushNotifications.php(159): Pusher\PushNotifications\PushNotifications->makeRequest('POST', '/publish_api/v1...', Array, Array)
#1 /Users/benjamintang/Documents/development/php/pure/send_notif/app.php(14): Pusher\PushNotifications\PushNotifications->publishToUsers(Array, Array)
#2 {main}
  thrown in /Users/benjamintang/Documents/development/php/pure/push-notifications-php-1/src/PushNotifications.php on line 65
```
I imported your lib by following this article, https://dev.to/aschmelyun/installing-a-local-composer-package-in-your-php-project-278k , and it added guzzlehttp/guzzle 7.4.5 in composer, so the dep resolution is ok i think. Actually it resolves the same version as our original 1.1.2 lib. If I rollback, it works fine, so I its not an issue with my php app.
I noticed you've changed how the $client property in class PushNotifications gets defined to public function __construct(private array $options, private GuzzleHTTP\Client|null $client = null) on line 21. Are you sure the $this->$client still exists after this point? Does it work for you?

**Comment by benjamin-tang-pusher** [comment]
The tests pass ok for me, and I've given the methods a quick go, so I'm passing this to the others for review. They will need to get the CI fully working.
It might be too early to drop support for 7.4 as well, but I will leave that to them.

**Comment by mischasigtermans** [comment]
Bump

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          ## ${{ env.VERSION }}
+          " >> CHANGELOG.tmp
+          grep "^*" xx01 >> CHANGELOG.tmp
+          grep -v "^# " CHANGELOG.md >> CHANGELOG.tmp
+          cp CHANGELOG.tmp CHANGELOG.md
+      - name: Prepare README
+        run: |
+          export MAJOR=$(echo "${{ env.VERSION }}" | cut -d'.' -f1)
+          export MINOR=$(echo "${{ env.VERSION }}" | cut -d'.' -f2)
+          sed -i "s|\"pusher/pusher-php-server\": \"^[0-9]*\.[0-9]*\"|\"pusher/pusher-php-server\": \"^${MAJOR}.${MINOR}\"|" README.md
```
Comment (amad) [inline .github/workflows/release.yml:66]: This step is wrong. line 66 - The README file on this project doesn't contain it. It's maybe better to not put version number in the README.

```suggestion
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      - name: Prepare Pusher.php
+        run: |
+          sed -i "s|public static \$VERSION = '[^']*'|public static \$VERSION = '${{ env.VERSION }}'|" src/Pusher.php
```
Comment (amad) [inline .github/workflows/release.yml:69]: There is also an error here. 

Presumably you want to update the version number in `src/PushNotifications.php` file which has a different format. https://github.com/pusher/push-notifications-php/blob/master/src/PushNotifications.php#L13

**Comment by benjamin-tang-pusher** [comment]
@SuperDJ you can just remove the release.yml from the PR. Automatic changelog/readme updating in the CI won't be necessary for this lib.

**Review by amad — APPROVED** [review]

**Comment by danharrin** [comment]
@amad @benjamin-tang-pusher I believe this repo still needs a new tag, so that the changes are made available on Packagist. Currently Beams is unusable with Laravel.

**Comment by SuperDJ** [comment]
@danharrin for Laravel it might still be required for [laravel-notification-channels/pusher-push-notifications](https://github.com/laravel-notification-channels/pusher-push-notifications) to update as well

**Comment by danharrin** [comment]
@SuperDJ the constraint for this package is `^1.1`, I assume this will be in a minor release?

**Pull request merged**